src/test/java/net/snowflake/ingest/streaming/internal/SnowflakeStreamingIngestChannelTest.java

```diff
@@ -808,6 +818,46 @@ public void testClose() throws Exception {
 
     // Calling close again on closed channel shouldn't fail
     channel.close().get();
+    Mockito.verify(client, Mockito.times(0)).dropChannel(Mockito.any());
+  }
+
+  @Test
+  public void testDropOnClose() throws Exception {
```

Comment:
could you add a test on drop on invalid?

Reply:
Done. Note that invalidation check is done at the start of the `close` function now and therefore, we never get to flushing or dropping the channel.